### PR TITLE
556 compatibility inquiry avaloniaui

### PR DESCRIPTION
**Pull request by vb2ae**
Add Avalonia ui support

### CODE REVIEW

**Comment by danwalmsley** [comment]
@vb2ae will release a preview 7 nuget package compatible with this to nuget in a day or so.

**Comment by vb2ae** [comment]
@danwalmsley thanks for the heads up. @danwalmsley and @Stannieman thanks for the help with this.

**Comment by danwalmsley** [comment]
@vb2ae is there a special nuget feed I can use to test these nuget packages produced on ci?

**Comment by vb2ae** [comment]
@danwalmsley I uploaded the package here for you.  https://www.nuget.org/packages/vb2ae.Caliburn.Micro.Avalonia/5.0.54-beta.ga5c370d704   but in general once the pr makes it into master we publish it to the myget feed right away

**Comment by vb2ae** [comment]
@danwalmsley I did notice while working on Features.Avalonia that I am getting an error  System.MissingMethodException: 'Method not found: 'System.Threading.Tasks.Task Avalonia.Threading.Dispatcher.InvokeAsync(System.Action, Avalonia.Threading.DispatcherPriority)'.'  when using the MessageBox.Avalonia.   Not sure if that is a problem with the messagebox.avalonia nuget package or related to the changes in Avalonia

**Comment by danwalmsley** [comment]
@vb2ae its because messagebox.avalonia havent yet updated to a compatible version. When I get preview7 out they will update and it will just work again.

**Comment by KasperSK** [comment]
What is the reason for the Features.Avalonia project not using the shared project for ViewModels? Is the point of that solution not to showcase that way of sharing ViewModels?

**Comment by vb2ae** [comment]
@KasperSK it is easier to debug. Looking at an issue with guard methods still.

File: src/Caliburn.Micro.Platform/XamlPlatformProvider.cs [hunk 1]
```diff
@@ -106,9 +116,14 @@ public class XamlPlatformProvider : IPlatformProvider {
                         exception = ex;
                     }
                 };
+#if AVALONIA
+                //TODO: (Avalonia) Need to check if GetAwaiter().GetResult() is a good solution here
+                dispatcher.InvokeAsync(action).GetAwaiter().GetResult(); 
+#else
                 dispatcher.Invoke(method);
+#endif
```
Comment (maxkatz6) [inline src/Caliburn.Micro.Platform/XamlPlatformProvider.cs:124]: In preview7 there is a dispatcher.Invoke method as well in Avalonia.

Comment (vb2ae) [reply src/Caliburn.Micro.Platform/XamlPlatformProvider.cs]: @maxkatz6 thanks made the change

**Review by maxkatz6 — APPROVED** [review]

File: src/Caliburn.Micro.Platform/ActionMessage.cs [hunk 2]
```diff
@@ -210,7 +210,8 @@ protected override void OnAttached()
                 if (View.ExecuteOnLoad(AssociatedObject, ElementLoaded))
                 {
 #if AVALONIA
-                    string eventName = "AttachedToLogicalTree";
+                    //string eventName = "AttachedToLogicalTree";
+                    string eventName = "Activated";
```
Comment (maxkatz6) [inline src/Caliburn.Micro.Platform/ActionMessage.cs]: In 11.0 Avalonia you can use Loaded events which works similarly how WPF does.

**Comment by Mrxx99** [comment]
Is there still something missing on this?

File: samples/features/Features.NetCore/Features.NetCore.csproj [hunk 3]
```diff
@@ -23,10 +23,10 @@
 
   <ItemGroup>
     <Reference Include="Caliburn.Micro.Core">
```
Comment (vb2ae) [inline samples/features/Features.NetCore/Features.NetCore.csproj:25]: should be release

File: samples/features/Features.WPF/Features.WPF.csproj [hunk 4]
```diff
@@ -47,7 +47,7 @@
       <HintPath>..\..\..\bin\Caliburn.Micro.Platform\release\net462\Caliburn.Micro.Platform.dll</HintPath>
     </Reference>
     <Reference Include="Caliburn.Micro.Platform.Core">
-      <HintPath>..\..\..\bin\Caliburn.Micro.Platform.Core\release\netstandard2.0\Caliburn.Micro.Platform.Core.dll</HintPath>
+      <HintPath>..\..\..\bin\Caliburn.Micro.Platform.Core\debug\netstandard2.0\Caliburn.Micro.Platform.Core.dll</HintPath>
```
Comment (vb2ae) [inline samples/features/Features.WPF/Features.WPF.csproj:50]: Should be release

File: samples/features/Features.Avalonia/Results/MessageDialogResult.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        public MessageDialogResult(string content, string title)
+        {
+            this.content = content;
+            this.title = title;
+        }
+
+        public override async void Execute(CoroutineExecutionContext context)
+        {
+            var dialog = MessageBoxManager
+                        .GetMessageBoxStandard("title", "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed...");
```
[the remark this answers is not in the document]
Comment (vb2ae) [reply samples/features/Features.Avalonia/Results/MessageDialogResult.cs]: fixed

File: samples/features/Features.Avalonia/ViewModels/ShellViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        protected override async Task OnInitializeAsync(CancellationToken cancellationToken)
+        {
+            await base.OnInitializeAsync(cancellationToken);
+
+            GoHome();
+        }
+
+        public void GoHome()
+        {
+            var menuVM = IoC.GetInstance(typeof(MenuViewModel), null) as MenuViewModel;
```
[the remark this answers is not in the document]
Comment (vb2ae) [reply samples/features/Features.Avalonia/ViewModels/ShellViewModel.cs]: fixed

**Pull request merged**